ARrehab/ARrehab/MovementGame.swift

```diff
@@ -47,15 +54,49 @@ class MovementGame : Minigame {
         self.total = num
         // For our purposes, we placed the player as a 2 centimeter sphere around the camera.
         self.playerCollisionEntity = TriggerVolume(shape: ShapeResource.generateSphere(radius: 0.01), filter: CollisionFilter(group:playerCollisionGroup, mask: targetCollisionGroup))
+        self.coachingState = .other
         super.init()
+        self.progress = [0, 0]
+        self.timer = Timer.scheduledTimer(withTimeInterval: 0.01, repeats: true) { timer in
```

Comment:
Because you are performing this action so rapidly, it may be better to use a DispatchQue.schedule() with the repeats option — Better for thread management

Reply:
ooh okay thanks

Reply:
Are you talking about a [DispatchSourceTimer](https://developer.apple.com/documentation/dispatch/dispatchsourcetimer) or a [DispatchQueue](https://developer.apple.com/documentation/dispatch/dispatchqueue/3329232-schedule)?

I can't seem to find much about repeated blocks for DispatchQueue on the internet for some reason.